ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -154,6 +155,12 @@ - (void)removeAllAnnotations
     [_annotations removeAllObjects];
 }
 
+- (void)deselectAnnotation
+{
+    NSLog(@"in deselect Annotation");
+    [self upsertAnnotation:_selectedAnnotation];
```

Comment:
Wait, so you're replacing the annotation in order to dismiss it? You can use `[_map deselectAnnotation:_selectedAnnotation animated: YES];` as per [the API docs](https://www.mapbox.com/ios-sdk/api/3.3.1/Classes/MGLMapView.html#/c:objc(cs)MGLMapView(im)deselectAnnotation:animated:)